lockapi.pm

```diff
@@ -29,9 +29,16 @@ sub _lock_action {
     my ($name, $where) = @_;
     my $param = {action => 'lock'};
     $param->{where} = $where if $where;
-    my $res = api_call('post', "mutex/$name", $param)->code;
-    return 1 if ($res == 200);
-
+    # try up to 7 times
+    my $res = "";
+  retry:
```

Comment:
because of limited perl knowlegde of @Soulofdestiny and me where we encountered at least once a loop and some statement inside which abort the whole outer function and not just the inner loop. See the "last" further down. To my understanding of [last](https://perldoc.perl.org/functions/last.html) we don't need the label here but I am scared… can you help us?